tests/modules/encoder_base_test.py

```diff
@@ -34,6 +34,26 @@ def setUp(self):
         self.sorting_indices = sorting_indices
         self.restoration_indices = restoration_indices
 
+    def test_non_stateful_states_are_sorted_correctly(self):
+        encoder_base = _EncoderBase(stateful=False)
+        initial_states = (Variable(torch.randn(6, 5, 7)),
+                          Variable(torch.randn(6, 5, 7)))
+        # Check that we sort the state for non-stateful encoders. To test
+        # we'll just use a "pass through" encoder, as we aren't actually testing
+        # the functionality of the encoder here anyway.
+        _, states, restoration_indices = encoder_base.sort_and_run_forward(lambda *x: x,
+                                                                           self.tensor,
+                                                                           self.mask,
+                                                                           initial_states)
+        zeros = torch.zeros([6, 2, 7])
+        for state, original in zip(states, initial_states):
+            assert list(state.size()) == [6, 3, 7]
```

Comment:
An explanation of what you're doing here would be helpful, as this is somewhat non-obvious.  Something like, "`sort_and_run_forward` strips fully-padded instances from the batch; in order to use the `restoration_indices` we need to add back the two that got stripped.  What we get back should match what we started with."